packages/sanity/src/core/form/inputs/ReferenceInput/useReferenceInfo.ts

```diff
@@ -66,15 +65,6 @@ export function useReferenceInfo(
         ),
       ),
     [getReferenceInfo, id, msg$, retry],
-    INITIAL_LOADING_STATE,
   )
-
-  // @todo test and see if this were fixed in `react-rx@2.1.x`
-  // workaround for a "bug" with useMemoObservable that doesn't
-  // return the initial value upon resubscription
-  const previousId = usePrevious(id, id)
-  if (id && previousId !== id) {
-    return INITIAL_LOADING_STATE
-  }
```

Comment:
@bjoerge can you verify that this is fixed?

Reply:
It should be, yes!